### PR TITLE
Be more forgiving when anything with opening a webbrowser fails

### PR DESCRIPTION
**Pull request by bartbroere**
This (hopefully) fixes an issue I encountered with Starboard Notebook. I'll link the issue here.

### CODE REVIEW

**Comment by jonmmease** [comment]
Hi @bartbroere, I'm not a maintainer of plotly projects any more, but FWIW these changes look reasonable to me!

File: packages/python/plotly/plotly/tests/test_io/test_renderers.py [hunk 1]
```diff
@@ -381,3 +382,27 @@ def test_repr_mimebundle_mixed_renderer(fig1):
...
         {"application/vnd.plotly.v1+json", "text/html"}
     )
+
+
+def test_missing_webbrowser_module(fig1):
+    """
+    Assert that no errors occur if the webbrowser module is absent
+    """
+    removed_webbrowser_module = sys.modules["webbrowser"]
+    del sys.modules["webbrowser"]
```
Comment (alexcjohnson) [inline packages/python/plotly/plotly/tests/test_io/test_renderers.py:392]: Does this actually cause `import webbrowser` to fail? Let's add something to the test to validate this:
```py
with pytest.raises(ImportError):
    import webbrowser
```

And if (as I think may be the case) it's still importable, here's a solution that looks promising: https://stackoverflow.com/a/60229056/9188800

**Comment by alexcjohnson** [comment]
@bartbroere the code change looks great, thanks! Just a couple of comments on the tests.

**Comment by bartbroere** [comment]
Thanks for the commit bringing it up to date. I might have some time tonight (GMT+2) to finish this PR and apply the feedback.

File: packages/python/plotly/plotly/tests/test_io/test_renderers.py (hunk 1 above)
Comment (bartbroere) [reply packages/python/plotly/plotly/tests/test_io/test_renderers.py]: This was helpful in pointing me in the right direction. Thanks! Landed on something similar buy slightly different in the end though.

**Comment by bartbroere** [comment]
@alexcjohnson I think it's ready for merge now

**Review by alexcjohnson — APPROVED** [review]
💃 Thanks @bartbroere, looks great!

**Pull request merged**